From linux-usb-owner@vger.kernel.org  Wed Jul 21 12:13:24 2010
Date: Mon, 12 Jul 2010 18:43:47 +0200
From: stephane duverger <stephane.duverger@gmail.com>
To: Greg KH <greg@kroah.com>
Cc: dbrownell@users.sourceforge.net, linux-usb@vger.kernel.org
Subject: USB: musb: forward debug mode feature to gadget
Message-ID: <20100712184347.12a4b627@riot.ssi.corp>

This is a patch for the musb usb controller.

It allows forwarding of the debug mode feature to its gadget in order
to be able to act as an ehci debug device.

This patch has been tested on an IGEPv2 board running a 2.6.35-rc1
kernel.

Signed-off-by: Stephane Duverger <stephane.duverger@gmail.com>
Cc: Felipe Balbi <felipe.balbi@nokia.com>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/usb/musb/musb_gadget_ep0.c |    3 +++
 1 file changed, 3 insertions(+)

--- a/drivers/usb/musb/musb_gadget_ep0.c
+++ b/drivers/usb/musb/musb_gadget_ep0.c
@@ -402,6 +402,9 @@ __acquires(musb->lock)
 					musb->g.a_alt_hnp_support = 1;
 					break;
 #endif
+				case USB_DEVICE_DEBUG_MODE:
+					handled = 0;
+					break;
 stall:
 				default:
 					handled = -EINVAL;
